JL.Core/Mining/MiningUtils.cs

```diff
@@ -440,6 +686,76 @@ public static async Task MineToFile(LookupResult lookupResult, string currentTex
         Utils.Logger.Information("Mined {PrimarySpelling}", lookupResult.PrimarySpelling);
     }
 
+    public static async ValueTask<bool[]?> CheckDuplicates(List<LookupResult> lookupResults, string currentText, int currentCharPosition)
+    {
+        List<Note> notes = [];
+        List<int> positions = [];
+        bool[] results = new bool[lookupResults.Count];
+
+        Dictionary<MineType, AnkiConfig>? ankiConfigDict = await AnkiConfig.ReadAnkiConfig().ConfigureAwait(false);
+        if (ankiConfigDict is null)
+        {
+            return null;
+        }
```

Comment:
This should be done at the very beginning of this method so that if we need to bail out we can do so before creating any unnecessary arrays/lists.